crates/rethnet_evm/src/db/fork.rs

```diff
@@ -0,0 +1,241 @@
+use hashbrown::HashMap;
+use revm::{db::DatabaseRef, Account, AccountInfo, Bytecode};
+
+use rethnet_eth::{Address, B256, U256};
+
+use crate::db::{
+    layered_db::{LayeredDatabase, RethnetLayer},
+    remote::{RemoteDatabase, RemoteDatabaseError},
+};
+
+struct ForkDatabase {
```

Comment:
Based on linguistics, should this be `ForkingDatabase` or `ForkedDatabase`?

Reply:
I'm happy to defer to your preference, but I felt that "forked database" sounds like you forked a database, but really you forked a node/chain, not a database, and it's your fork (noun, not past tense verb) of that node/chain that's serving as the backing for the database.